### PR TITLE
Minor cleanups to geoip and lmdb

### PR DESCRIPTION
**Pull request by fredmorcos**
### Short description
Cleanup some compilation warnings in geoip and lmdb.

### Checklist
I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: modules/geoipbackend/geoipbackend.cc [hunk 1]
```diff
@@ -121,7 +121,7 @@ static bool validateMappingLookupFormats(const vector<string>& formats)
   return true;
 }
 
-bool GeoIPBackend::loadDomain(const YAML::Node& domain, unsigned int id, GeoIPDomain& dom)
+bool GeoIPBackend::loadDomain(const YAML::Node& domain, unsigned int /* id */, GeoIPDomain& dom)
```
Comment (rgacogne) [inline modules/geoipbackend/geoipbackend.cc:124]: If we don't use `id` it should probably go?

**Review by rgacogne — CHANGES_REQUESTED** [review]

File: modules/geoipbackend/geoipbackend.cc (hunk 1 above)
Comment (fredmorcos) [reply modules/geoipbackend/geoipbackend.cc:124]: Correct

**Review by Habbie — DISMISSED** [review]

**Comment by rgacogne** [comment]
I'm sorry, Fred, but I now realize that the change I asked you to do is wrong. The problem is that we need to `dom.id` to a different value for all domains, and it looks like we were no longer doing that since https://github.com/PowerDNS/pdns/commit/4287cb762b1d3603047d60e3d0918e7cf09ddd00 So I think the correct fix is to revert the last change and to set `dom.id = id` as we used to do before the aforementioned commit.

**Review by Habbie — APPROVED** [review]

**Review by rgacogne — APPROVED** [review]
:+1:

File: modules/geoipbackend/geoipbackend.cc [hunk 2]
```diff
@@ -51,9 +52,9 @@ struct GeoIPService
 
 struct GeoIPDomain
 {
-  int id;
+  std::size_t id{};
```
Comment (fredmorcos) [inline modules/geoipbackend/geoipbackend.cc:55]: Is this OK?

Comment (Habbie) [reply modules/geoipbackend/geoipbackend.cc:55]: it needs to fit in 31 bits, so uint32_t might make more sense (that's what dnsbackend.hh uses)

**Comment by fredmorcos** [comment]
This needs another review.

**Review by rgacogne — APPROVED** [review]

**Review by Habbie — APPROVED** [review]

**Comment by Habbie** [comment]
I believe this fixes #12876 !

**Comment by fredmorcos** [comment]
> I believe this fixes #12876 !

Yes it does.

**Pull request merged**